independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/InterceptorBindingRegistrar.java

```diff
@@ -1,15 +1,98 @@
 package io.quarkus.arc.processor;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
 import java.util.Map;
+import java.util.Map.Entry;
 import java.util.Set;
+import java.util.function.Predicate;
 import org.jboss.jandex.DotName;
 
 public interface InterceptorBindingRegistrar extends BuildExtension {
 
     /**
      * Annotations in a form of {@link DotName} to be considered interceptor bindings.
      * Optionally, mapped to a {@link Collection} of non-binding fields
+     * 
+     * @deprecated Use {@link #getAdditionalBindings()} instead.
      */
-    Map<DotName, Set<String>> registerAdditionalBindings();
+    @Deprecated(forRemoval = true)
+    default Map<DotName, Set<String>> registerAdditionalBindings() {
+        return Collections.emptyMap();
+    }
+
+    /**
+     * @return the list of additional interceptor bindings
+     */
+    default List<InterceptorBinding> getAdditionalBindings() {
+        Map<DotName, Set<String>> map = registerAdditionalBindings();
+        if (map.isEmpty()) {
+            return Collections.emptyList();
+        }
+        List<InterceptorBinding> bindings = new ArrayList<>();
+        for (Entry<DotName, Set<String>> e : map.entrySet()) {
+            bindings.add(InterceptorBinding.of(e.getKey(), e.getValue()));
+        }
+        return bindings;
+    }
+
+    /**
+     * Represents an additional interceptor binding definition. The name is used to identify an annotation class and the
+     * {@link #test(String)} method is used to match annotation values that should be considered non-binding as defined by the
+     * CDI spec.
+     */
+    interface InterceptorBinding extends Predicate<String> {
+
+        static InterceptorBinding of(Class<?> clazz) {
+            return of(DotName.createSimple(clazz.getName()));
+        }
+
+        static InterceptorBinding of(DotName name) {
+            return new InterceptorBinding() {
+
+                @Override
+                public boolean test(String memberName) {
+                    return false;
+                }
+
+                @Override
+                public DotName getName() {
+                    return name;
+                }
+            };
+        }
+
+        static InterceptorBinding of(Class<?> clazz, Predicate<String> predicate) {
+            return of(DotName.createSimple(clazz.getName()), predicate);
+        }
+
+        static InterceptorBinding of(DotName name, Predicate<String> predicate) {
+            return new InterceptorBinding() {
+
+                @Override
+                public boolean test(String memberName) {
+                    return predicate.test(memberName);
+                }
+
+                @Override
+                public DotName getName() {
+                    return name;
+                }
+            };
+        }
+
+        static InterceptorBinding of(DotName name, Set<String> nonbinding) {
```

Comment:
How about adding `of(Class<?> clazz, Set<String> nonbinding)`? Frankly, I'd assume that would be the most common form, because you usually know upfront the class and the set of non-binding members.

Reply:
I think that the most common form will be `InterceptorBinding.of(name, () -> true)` to include all members which is also the reason why I started to work on this PR ;-) but yes, I can ofc add this method.

Reply:
I'd personally use `InterceptorBinding.of(name, Set.of())`, but that's just me :-)

Reply:
I think that's not correct - `InterceptorBinding.of(name, Set.of())` means that `none` is nonbinding,i.e. all are binding...

Reply:
I'm always here for you! :D